### PR TITLE
Beginnings of a Vue js sample application

### PR DESCRIPTION
**Pull request by aullman**
I was playing around a bit with getting Vuejs to work with OpenTok and thought it would make a good addition to this repo. WDYT @jeffswartz @msach22?

If you think it makes sense I can add a README.

### CODE REVIEW

**Comment by aullman** [comment]
@leahciMic I'd also be interested to get your feedback on this. Not sure if I'm doing things the best way.

File: Vue-Basic-Video-Chat/js/app.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+      type: OT.Session,
+      required: false
+    },
+    opts: {
+      type: Object,
+      required: false
+    }
+  },
+  mounted: function() {
+    publisher = OT.initPublisher(this.$el, this.opts, (err) => {
```
Comment (aiham) [inline Vue-Basic-Video-Chat/js/app.js]: Should this variable be declared?

File: Vue-Basic-Video-Chat/js/app.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        this.$emit('error', err);
+      } else {
+        this.$emit('publisherCompleted');
+      }
+    });
+    this.$emit('publisherCreated', publisher);
+    if (this.session && this.session.isConnected()) {
+      this.session.publish(publisher);
+    } else {
+      this.session.on('sessionConnected', () => {
```
Comment (aiham) [inline Vue-Basic-Video-Chat/js/app.js]: Can this be `this.session.once`?

File: Vue-Basic-Video-Chat/js/app.js (hunk 1 above)
Comment (aullman) [reply Vue-Basic-Video-Chat/js/app.js]: Yes, good catch. I also added an error handler for session.publish

File: Vue-Basic-Video-Chat/js/app.js (hunk 2 above)
Comment (aullman) [reply Vue-Basic-Video-Chat/js/app.js]: I think it should be on. If you disconnect and connect again we will want to publish the publisher again.

File: Vue-Basic-Video-Chat/src/Session.vue [hunk 3]
```diff
@@ -0,0 +1,72 @@
+<template>
+  <div id="session" v-on:error="errorHandler">
```
Comment (leahciMic) [inline Vue-Basic-Video-Chat/src/Session.vue]: Not a huge deal, but `@` is a convenient shorthand for `v-on`, in the same way `:` is a shorthand for `v-bind`

File: Vue-Basic-Video-Chat/src/Session.vue [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+      data.streams.push(event.stream);
+    });
+    data.session.on('streamDestroyed', (event) => {
+      const idx = data.streams.indexOf(event.stream);
+      if (idx > -1) {
+        data.streams.splice(idx, 1);
+      }
+    });
+  },
+  data: () => data,
```
Comment (leahciMic) [inline Vue-Basic-Video-Chat/src/Session.vue]: Typically in a reusable component, you'll want to return a new object from data each time. If you return the same object (identity), then each component will share the same state.

**Comment by aullman** [comment]
@leahciMic Thanks for the feedback, I made the changes you suggested.

For the main.js file do you mean it should be a .vue file? I was just copying the style of the template https://github.com/vuejs-templates/webpack-simple/blob/master/template/src/main.js

the webpack template also seems to do the same thing https://github.com/vuejs-templates/webpack/blob/develop/template/src/main.js

Where it has a render function which renders the base application component.

**Pull request merged**